src/main/java/com/mongodb/hibernate/internal/type/MongoStructJdbcType.java

```diff
@@ -105,19 +106,19 @@ public EmbeddableMappingType getEmbeddableMappingType() {
      * {@link #createJdbcValue(Object, WrapperOptions)} is not called by Hibernate ORM.
      */
     @Override
-    public BsonValue createJdbcValue(@Nullable Object domainValue, WrapperOptions options) {
+    public BsonDocument createJdbcValue(@Nullable Object domainValue, WrapperOptions options) {
         throw fail();
     }
 
-    private BsonValue createBsonValue(@Nullable Object domainValue, WrapperOptions options) throws SQLException {
+    private @Nullable BsonDocument createBsonValue(@Nullable Object domainValue, WrapperOptions options)
+            throws SQLException {
         if (domainValue == null) {
-            throw new FeatureNotSupportedException(
-                    "TODO-HIBERNATE-48 https://jira.mongodb.org/browse/HIBERNATE-48 return toBsonValue(domainValue)");
+            return null;
```

Comment:
we have to return `null` instead of `BsonNull` for we need to return domain value for `bindValue`; Struct -> BsonDocument transformation should be the only exception; for that reason, I changed the method return type to better align with its purpose.